kratos/python_scripts/python_solver.py

```diff
@@ -33,7 +33,9 @@ def __init__(self, model, settings):
 
         # TODO remove the check once the derived solvers implement this
         if hasattr(self, '_validate_settings_in_baseclass'):
-            self.settings.ValidateAndAssignDefaults(self.GetDefaultSettings())
+            default_settings = self.GetDefaultSettings()
```

Comment:
@philbucher This is the only change concerning core, I did it in order to apply the recursive validation

Reply:
I think that we cannot use recursive validation here
I tried this one in structural and that basically broke selection of the linear solver

reason is that if different linear solvers have different sub-settings then this breaks it

You could do the recursive one on top for specific subparameters if you need it, at least thats what I had in mind

Reply:
That's why I use first RecursivelyAddMissingParameters

Reply:
![](https://media.giphy.com/media/d3mlE7uhX8KFgEmY/giphy.gif)

Reply:
but does it work if e.g. your linear solver settings contain "iter_num", which is not in your defaults?

Reply:
Yes, because I am transferring from the given settings, will add the missing ones on the defaults, but will not override the given settings

Reply:
hm ok then let me test as well if what I have in mind works

Reply:
Ok